### PR TITLE
Add wiki page for Horizons tool

### PR DESCRIPTION
**Pull request by Roxeena**

### CODE REVIEW

File: docs/builders/ephemeris/horizons-gui.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: Download Horizons data via OpenSpace
+layout: default
+
+grand_parent: Builders
+parent: Ephemeris
+nav_order: 1
+---
+
+Since version 0.18.0 of OpenSpace there is a new feature that allows our users to access Horizons data directly via OpenSpace with no need to go to the [Horizons website](https://ssd.jpl.nasa.gov/horizons.cgi). This feature can be found in the OpenSpace launcher by pressing a series of buttons:
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md:10]: "...directly via OpenSpace with no need to go to the [Horizons website]" -> "directly via OpenSpace, without needing to go to the [Horizons website]"

Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "...directly via OpenSpace with no need to go to the [Horizons website]" -> "directly via OpenSpace, without needing to go to the [Horizons website]"

File: docs/builders/ephemeris/horizons-gui.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+2. Press the *Edit* button close to the asset selection
+3. In the top right corner of the new window press the *New Asset* button
+4. Finally press the *Generate Horizons File* button
+5. At this point you should see the window shown in the image below
+
+![](images/gui.png)
+
+
+# The Horizons Tool
+In the new window there is a few places where you can input specifications for what Horizons data you want to access. Once you have put in the input you can press the *Save* button at the bottom of the window. At this point OpenSpace will ask the Horizons service for the data you want and download it and store it on your computer. This wiki will go through every step of this process and explain what each field in the window means. This process is very similar to how you would do it on the Horizons web page that you can read more about [here](horizons-web).
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "Once you have put in the input you can press the *Save* button at the bottom of the window" -> "Once you have entered all information you can press the *Save* button at the bottom of the window"

Comment (WeirdRubberDuck) [reply docs/builders/ephemeris/horizons-gui.md]:
```
This process is very similar to how you would do it on the Horizons web page that you can read more about [here](horizons-web).
```
->
```
This process is very similar to how you would do it on the Horizons web page. Read more about that [here](horizons-web).
```

(obs, switched commenting style from here on)

File: docs/builders/ephemeris/horizons-gui.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## Horizons data type
+Here you can select what format you want your data in, OpenSpace supports both **Vector Table** and **Observer Table**. However, we recommend the **Vector Table** format since this will give a better match between your data and the rest of OpenSpace.
+
+## File Path
+Here you specify where on your computer you want the data to be stored once it has been downloaded. You can click the *Browse* button to browse with the file explorer to specify where the data should be stored. This is required to be specified for the data to be downloaded. If there was any issue with the download, then there will instead be an error file stored at this location. The error file will be named <code>XXX_error.txt</code> where <code>XXX</code> is the name of the file you specified. If you adjust your input to fix this issue and the download succeeds this time, this error file will be deleted automatically.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: The first bit here seems a bit wordy to me. Maybe just shorten the sentence: "You can click the *Browse* button to browse with the file explorer to specify where the data should be stored."?

File: docs/builders/ephemeris/horizons-gui.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Target Body
+Here you can type in the name of the thing you want data for. For example if you want data for the Voyager 1 spacecraft you can write <code>Voyager 1</code> here. Horizons have data for many different things from spacecraft, planets, moons, comets and asteroids to lagrange points and more. In this field and the **Observer Location** field the characters <code>¤ < > § £ ´ ¨ €</code> and other special characters from other languages are not allowed. Additionally the characters <code>' =</code> have some limitations in how they can be used.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: I would rephrase so that not every section starts with "Here you...". Maybe "in this field". or just skip it completely where it doesn't seem needed?

Comment (WeirdRubberDuck) [reply docs/builders/ephemeris/horizons-gui.md]: Lin break before "In this field and the **Observer Location** field" ?

Comment (WeirdRubberDuck) [reply docs/builders/ephemeris/horizons-gui.md]: Also, "thing" sounds a bit vague. How about "object"? :)  Some improvement suggestion

"Here you can type in the name of the thing you want data for. For example if you want data for..."
->
"Here you specify the name of the object you want to get data for. For example, if you want data for..."

Comment (WeirdRubberDuck) [reply docs/builders/ephemeris/horizons-gui.md]: Actually: 
`"Here you specify the name of the object you want to get data for. For example, if you want data for the Voyager 1 spacecraft you can write <code>Voyager 1</code> here. Horizons have data for many different things from spacecraft, planets, moons, comets and asteroids to Lagrange points and more. 

In this and the **Observer Location** field the characters <code>¤ < > § £ ´ ¨ €</code> and other special characters from other languages are not allowed. Additionally, the characters <code>' =</code> have some limitations in how they can be used."

Might also be good to specify that "Voyager 1" must match an object in the Horizons framework and can't just be any name (right?)

File: docs/builders/ephemeris/horizons-gui.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+## Observer Location
+Here you can type in the name of the reference point you want for the data. For example if you want to show the movement of the spacecraft Voyager 1 over time, the reference point should probably be the Sun or the Solar System Barycenter. But if you instead want to show a satellite that orbits around Mars, the reference point should probably be Mars.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "Here you specify the name of the reference point you want for the data. For example, if you want to show the movement of the spacecraft Voyager 1 over time, the reference point should probably be the Sun or the Solar System Barycenter. But if you instead want to show a satellite that orbits around Mars, the reference point should probably be Mars."

File: docs/builders/ephemeris/horizons-gui.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+## Start Time
+Here you specify the start of the time range you want data for.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "The start of the time range you want data for."

An example f how to get rid of the "Here you.." repetitiveness. It's not really neede dhere
Could do the same for stop time

File: docs/builders/ephemeris/horizons-gui.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+5. At this point you should see the window shown in the image below
+
+![](images/gui.png)
+
+
+# The Horizons Tool
+In the new window there is a few places where you can input specifications for what Horizons data you want to access. Once you have put in the input you can press the *Save* button at the bottom of the window. At this point OpenSpace will ask the Horizons service for the data you want and download it and store it on your computer. This wiki will go through every step of this process and explain what each field in the window means. This process is very similar to how you would do it on the Horizons web page that you can read more about [here](horizons-web).
+
+## Horizons data type
+Here you can select what format you want your data in, OpenSpace supports both **Vector Table** and **Observer Table**. However, we recommend the **Vector Table** format since this will give a better match between your data and the rest of OpenSpace.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: Remove "can". "Here you select...."

This "can" can be removed from almost every section

File: docs/builders/ephemeris/horizons-gui.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+3. In the top right corner of the new window press the *New Asset* button
+4. Finally press the *Generate Horizons File* button
+5. At this point you should see the window shown in the image below
+
+![](images/gui.png)
+
+
+# The Horizons Tool
+In the new window there is a few places where you can input specifications for what Horizons data you want to access. Once you have put in the input you can press the *Save* button at the bottom of the window. At this point OpenSpace will ask the Horizons service for the data you want and download it and store it on your computer. This wiki will go through every step of this process and explain what each field in the window means. This process is very similar to how you would do it on the Horizons web page that you can read more about [here](horizons-web).
+
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md:24]: Add a sentence: 

"The following sections describe the different parts of the Horizons UI. "

or something? För "röd tråd"?

File: docs/builders/ephemeris/horizons-gui.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+Here you can type in the name of the reference point you want for the data. For example if you want to show the movement of the spacecraft Voyager 1 over time, the reference point should probably be the Sun or the Solar System Barycenter. But if you instead want to show a satellite that orbits around Mars, the reference point should probably be Mars.
+
+## Start Time
+Here you specify the start of the time range you want data for.
+
+## Stop Time
+Here you specify the end of the time range you want data for.
+
+## Step Size
+Here you can specify at what resolution you want the data to be in. You start with a number and then you can choose a time unit in the drop down menu to the right.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "The step size is the temporal resolution you want the data to have." 

And maybe an example, but the last sentence is not necessary :)

File: docs/builders/ephemeris/horizons-gui.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+## Horizons log messages
+This is a log that only displays messages related to your input in this tool and the download of the data. When using this tool there are a few different issues that you can encounter and they will be explained in more detail below. All of the issues you will encounter will be detailed in this log and the messages will guide you towards a fix. If OpenSpace fails with getting this helpful information there will be an error file stored at the specified location in the **File Path** field. This file might contain some useful information about what went wrong and how it can be fixed.
+
+
+# Issues and Solutions
+There are a few different things that can go wrong while using this tool. Most of the issues are related to the input and can be fixed with some adjustments. The particular parameters that need to be adjusted will be highlighted in red and there will be some messages in the Horizons log that can give you tips on how to fix it.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: Add a sentence, för "röd tråd"

"Some common errors and solutions for those errors are explained below"

File: docs/builders/ephemeris/horizons-gui.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+## No matching target
+The target you searched for could not be found. Check that the name has been spelt correctly and you can also add a <code>*</code> at the end of the name to widen your search. If you still cannot find what you are looking for then it is possible that there might be an alternative name for it in the Horizons database or that it does not exist.
+
+## Multiple matching targets
+The target you searched for found multiple matches in the Horizons database. There should be a added drop down menu close to the **Target Body** field where you can select one of the matches. If none of them is what you were looking for, you can keep the first item in the list selected and write a new name to search for.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "There should be a added drop down menu ..."
->
"There should be an added drop-down menu ..."

File: docs/builders/ephemeris/horizons-gui.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+## Target and Observer are the same
+This issue only happens when the **Observer Table** format is selected and happens when the target and the observer are the same object. This can be fixed by either changing the observer or the target to something else. If the **Vector Table** format is selected, then this is allowed even if it does not give any useful results (all the data will be zero).
+
+## No matching observer
+The observer location you searched for could not be found. Check that the name has been spelt correctly and you can also add a <code>@</code> to the beginning of the name to widen your search. If you still cannot find what you are looking for then it is possible that there might be an alternative name for it in the Horizons database or that it does not exist.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: Split last sentence into two subsentences: 

"If you still cannot find what you are looking for, it is possible that there might be an alternative name for it in the Horizons database or that it does not exist."
?

File: docs/builders/ephemeris/horizons-gui.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+## Multiple matching observers
+The observer you searched for found multiple matches in the Horizons database. There should be a added drop down menu close to the **Observer Location** field where you can select one of the matches. If none of them is what you were looking for, you can keep the first item in the list selected and write a new name to search for.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "The observer you searched for found multiple matches in the Horizons database."
->
"Multiple matches were found in the Horizons database for the observer you searched for."  ?

Comment (WeirdRubberDuck) [reply docs/builders/ephemeris/horizons-gui.md]: "There should be a added drop down menu ..."
->
"There should be an added drop-down menu ..."

File: docs/builders/ephemeris/horizons-gui.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+## Multiple matching observer stations
+The observer you searched for found multiple matching observer stations in the Horizons database. There should be a added drop down menu close to the **Observer Location** field where you can select one of the matches. If none of them is what you were looking for, you can keep the first item in the list selected and write a new name to search for. If you did not intend to search for an observer station, try to add a <code>@</code> to the beginning of the name to widen your search.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: Same comment as above

File: docs/builders/ephemeris/horizons-gui.md [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+
+## Time range is outside available time range for target
+Each target has data available in a certain time range and this issue happens when you ask for data outside of that range. There should be a added button *Import Time Range* close to the time selection fields. This button will set the time selection to the entire available range for the current target. There will also be a message in the Horizons log detailing what this range is for the target. This can be helpful in adjusting the time selection to be within the range. If the button or the message does not show up, then there will be an error file stored at the specified location in the **File Path** field and this file might contain some useful information on how to fix the issue.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: "There should be a added button..." -> "There should be an added button..."

File: docs/builders/ephemeris/horizons-gui.md [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+## The projected size of the data is too large
+This problem arises when you ask for data over a very long time span and at the same time have a very fine resolution. Horizons data have a limit on how big the files can be and if the output is projected to exceed that limit, this issue occurs. To fix this issue you need to either shorten the time span or decrease the resolution, in some cases it might be necessary to do both. You can also separate your request into several files. For example first you can ask for the first 10 years of data into one file and then ask for the remaining 10 years of data in a second file. These files can then be combined in OpenSpace to show all of the data without sacrificing the resolution.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md]: This problem arises when you ask for data over a very long time span and at the same time have a very fine resolution. Horizons data have a limit on how big the files can be and if the output is projected to exceed that limit, this issue occurs. To fix this issue you need to either shorten the time span or decrease the resolution. In some cases, it might be necessary to do both. You can also separate your request into several files. For example, first you can ask for the first 10 years of data in one file and then ask for the remaining 10 years of data in a second file. These files can then be combined in OpenSpace to show all of the data without sacrificing the resolution.

File: docs/builders/ephemeris/horizons-web.md [hunk 17]
```diff
@@ -0,0 +1,150 @@
+---
+title: Download Horizons data from website
+layout: default
+
+grand_parent: Builders
+parent: Ephemeris
+nav_order: 1
+---
+
+This covers the basics of downloading ephemeris data for a solar system object from JPL Horizons webpage in the proper format for OpenSpace, and then configuring a renderable asset for it.  This document only applies to an object in the solar system, using the sun as the origin.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-web.md]: "the sun" -> "the Sun"

File: docs/builders/ephemeris/horizons-web.md [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+- object page = enabled
+
+## Generate Output
+Finally, click on the **Generate Ephemeris** button to get the ephemeris data in text format. Then click the button **Download Results** and save it to a **.hrz** file.
+
+# Examine Downloaded Output
+Examine the .hrz file and verify that it is valid.  If it only contains the header but no space-delimited lines of data, then there should be an error message explaining the problem.  For example, this will happen if the start/end dates are beyond the valid time range of the target object.
+
+# Data Usage
+This wiki covers the scenario where an asset file references a data file served by one of the OpenSpace sync servers.  At runtime, the file will be downloaded if a local copy does not already exist.  This requires the file generated in the above steps to be uploaded to an OpenSpace sync server by one of the project's server administrators.
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-web.md]: "in the above steps" -> "in the steps above" ? maybe nitpicky

File: docs/builders/ephemeris/index.md [hunk 19]
```diff
@@ -10,6 +10,7 @@ nav_order: 2
 
 Pages that deal with different types of Ephemeres that are supported:
 
-- [Using JPL Horizons Ephemeris for Solar System Objects](horizons)
+- [Using the GUI in OpenSpace to download Horizons data](horizons-gui)
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/index.md]: Should be big Ds:    "Using the GUI in OpenSpace to Download Horizons Data" to match the format of the other page titles

File: docs/builders/ephemeris/horizons-gui.md [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+Here you specify the end of the time range you want data for.
+
+## Step Size
+Here you can specify at what resolution you want the data to be in. You start with a number and then you can choose a time unit in the drop down menu to the right.
+
+## Horizons log messages
+This is a log that only displays messages related to your input in this tool and the download of the data. When using this tool there are a few different issues that you can encounter and they will be explained in more detail below. All of the issues you will encounter will be detailed in this log and the messages will guide you towards a fix. If OpenSpace fails with getting this helpful information there will be an error file stored at the specified location in the **File Path** field. This file might contain some useful information about what went wrong and how it can be fixed.
+
+
+# Issues and Solutions
```
Comment (WeirdRubberDuck) [inline docs/builders/ephemeris/horizons-gui.md:52]: Nitpicky, but you vary what title casing you use. This should be "Issues and solutions" to match the ones below. Maybe look through them again and decide on a style?

**Review by WeirdRubberDuck — CHANGES_REQUESTED** [review]

File: docs/builders/ephemeris/horizons-gui.md (hunk 4 above)
Comment (Roxeena) [reply docs/builders/ephemeris/horizons-gui.md]: What do you mean that Voyager 1 has to be in the Horizons framework? This works more or less like a search and if there is no match there is an error that is explained further down.

Comment (WeirdRubberDuck) [reply docs/builders/ephemeris/horizons-gui.md]: I just thought that the "search" part could be worth clarifying here already. Maybe something like: 

"Here you specify the name of the object you want to get data for.  It is used as a search string and has to match something that exists in Horizons to get a successful result. "

But maybe that's "dumbifying" it, I don't know

Comment (Roxeena) [reply docs/builders/ephemeris/horizons-gui.md]: Since I have a section further down on how to solve that problem if it arises, I think I will leave it as it is for now.

**Review by WeirdRubberDuck — APPROVED** [review]
